lib/base-text-component.js

```diff
@@ -23,21 +28,62 @@ export default class BaseTextComponent extends Component {
 
     updateValue(text) {
         let self = this;
+        let currSelection = self.state.selection;
+        let newSelection = currSelection;
```

Comment:
this will be a reference to self.state.selection. 
you can use Object.assign({}, currSelection) or ...currSelection
if you want to make a copy of currSelection